src/instana/instrumentation/pika.py

```diff
@@ -2,166 +2,219 @@
-                    logger.debug("basic_get_with_instana: ", exc_info=True)
+                    _extract_consumer_tags(span, conn=instance.connection, queue=queue)
+                except Exception:
+                    logger.debug("pika basic_get_with_instana error: ", exc_info=True)
 
                 try:
                     callback(channel, method, properties, body)
-                except Exception as e:
-                    scope.span.log_exception(e)
+                except Exception as exc:
+                    span.record_exception(exc)
                     raise
 
         args = (queue, _cb_wrapper) + args
         return wrapped(*args, **kwargs)
 
-    @wrapt.patch_function_wrapper('pika.adapters.blocking_connection', 'BlockingChannel.basic_consume')
-    def basic_consume_with_instana(wrapped, instance, args, kwargs):
+    @wrapt.patch_function_wrapper(
+        "pika.adapters.blocking_connection", "BlockingChannel.basic_consume"
+    )
+    def basic_consume_with_instana(
+        wrapped: Callable[..., Any],
+        instance: pika.adapters.blocking_connection.BlockingChannel.basic_consume,
+        args: Tuple[int, str, Tuple[Any, ...]],
+        kwargs: Dict[str, Any],
+    ) -> object:
         def _bind_args(queue, on_message_callback, *args, **kwargs):
             return (queue, on_message_callback, args, kwargs)
 
         queue, on_message_callback, args, kwargs = _bind_args(*args, **kwargs)
 
         def _cb_wrapper(channel, method, properties, body):
-            parent_span = tracer.extract(opentracing.Format.HTTP_HEADERS, properties.headers,
-                                         disable_w3c_trace_context=True)
+            parent_context = tracer.extract(
+                Format.HTTP_HEADERS, properties.headers, disable_w3c_trace_context=True
+            )
 
-            with tracer.start_active_span("rabbitmq", child_of=parent_span) as scope:
+            with tracer.start_as_current_span(
+                "rabbitmq", span_context=parent_context
+            ) as span:
                 try:
-                    _extract_consumer_tags(scope.span,
-                                           conn=instance.connection._impl,
-                                           queue=queue)
-                except:
-                    logger.debug("basic_consume_with_instana: ", exc_info=True)
+                    _extract_consumer_tags(
+                        span, conn=instance.connection._impl, queue=queue
+                    )
+                except Exception:
+                    logger.debug(
+                        "pika basic_consume_with_instana error:", exc_info=True
+                    )
 
                 try:
                     on_message_callback(channel, method, properties, body)
-                except Exception as e:
-                    scope.span.log_exception(e)
+                except Exception as exc:
+                    span.record_exception(exc)
                     raise
```

Comment:
raise keyword will record another error in span.

```suggestion
```

Reply:
Done